src/extensions/extension-loader.ts

```diff
@@ -30,18 +31,39 @@ export class ExtensionLoader {
         })
       });
     }
+    this.manageExtensionsState();
   }
 
-  @computed get userExtensions(): LensExtension[] {
-    return [...this.instances.values()].filter(ext => !ext.isBundled)
+  @computed get userExtensions(): InstalledExtension[] {
+    return Array.from(this.toJSON().values()).filter(ext => !ext.isBundled)
   }
 
-  async init() {
-    const { extensionManager } = await import("./extension-manager");
-    const installedExtensions = await extensionManager.load();
-    this.extensions.replace(installedExtensions);
+  protected async manageExtensionsState() {
+    await extensionsStore.whenLoaded;
+    await when(() => this.isLoaded);
+
+    // apply initial state
+    this.extensions.forEach((ext, extId) => {
+      ext.enabled = ext.isBundled || extensionsStore.isEnabled(extId);
```

Comment:
nvm, this should be done purely in the `extensionsStore`. I don't see any reason why it shouldn't know that entire truth. I also see a problem here where there is a disconnect between `this.extensions` and `extensionStore` for the enabled flag. There should be one source of truth.